packages/messaging/src/ui/components/ContentCardView/ContentCardView.tsx

```diff
@@ -167,7 +153,18 @@ export const ContentCardView: React.FC<ContentViewProps> = ({
     }
   }, [_styleOverrides, cardVariant]);
 
-  return (
+  // Call listener on mount to signal view display (only once and only when visible)
+  useEffect(() => {
+    if (isVisible && !isDisplayedRef.current) {
+      listener?.('onDisplay', template);
+      // Track display event using propositionItem
+      template.track?.(MessagingEdgeEventType.DISPLAY);
+      isDisplayedRef.current = true;
+    }
+  }, [isVisible, listener, template]);
+
+  // Use conditional rendering instead of early return to avoid hooks issues
+  return !isVisible ? null : (
```

Comment:
instead of the early return you can just move the deleted block above the return, after the hooks and it reads a bit cleaner. Not sure how they got placed after the hooks.